code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -103,9 +104,11 @@
 	SetInitDirections()
 	var/obj/machinery/atmospherics/node = nodes[1]
 	if(node)
-		node.disconnect(src)
+		if(src in node.nodes) //Only if it's actually connected. On-pipe version would is one-sided.
+			node.disconnect(src)
 		nodes[1] = null
-	nullifyPipenet(parents[1])
+	if(parents[1])
```

Comment:
wont this runtime on an empty list?  would length(parents) be better here?

Reply:
parents is always device element length, the issue is it's null not the length.